server/util/grpc_client/grpc_client.go

```diff
@@ -78,6 +82,12 @@ func CommonGRPCClientOptions() []grpc.DialOption {
 
 			// If true, client sends keepalive pings even with no active RPCs.
 			PermitWithoutStream: true,
+
+			PingCallback: func(connectionID uint64) {
+				if *logKeepalivePings {
+					log.Debugf("Sending keepalive ping for connection %d", connectionID)
```

Comment:
do you want this to be at a higher level than debug so we can see it in prod logs?

Reply:
this is happening pretty frequently in dev (tens of times per day) so I think it's fine to just have it at debug level